packages/zosuss/src/SshBaseHandler.ts

```diff
@@ -58,22 +59,103 @@ export abstract class SshBaseHandler implements ICommandHandler {
+                                sshSessCfgWithCreds,
+                                commandParameters.arguments,
+                                {
+                                    parms: commandParameters,
+                                    propertyOverrides: sshSessCfgOverride,
+                                    supportedAuthTypes: [
+                                        SessConstants.AUTH_TYPE_BASIC,
+                                    ],
+                                    propsToPromptFor: [
+                                        {
+                                            name: "keyPassphrase",
+                                            isGivenValueValid: (givenValue: string) => {
+                                                let saveKP: boolean = true;
+                                                const result = utils.parseKey(
+                                                    fs.readFileSync(
+                                                        sshSessCfgWithCreds.privateKey
+                                                    ),
+                                                    givenValue
+                                                );
+                                                if (result instanceof Error)
+                                                    saveKP =
+                                                        !result.message.includes(
+                                                            "no passphrase given"
+                                                        ) &&
+                                                        !result.message.includes(
+                                                            "bad passphrase"
+                                                        );
+                                                return saveKP;
+                                            },
+                                        },
+                                    ],
+                                }
+                            );
+                        this.mSession = new SshSession(sshSessCfgWithCreds);
+                        await this.processCmd(commandParameters);
+                        success = true;
+                    } catch (retryError) {
+                        this.console.log(
+                            "\n" +
+                                `Key passphrase authentication failed! (${
+                                    ++attempt
+                                }/${maxAttempts})` +
+                                "\n"
+                        );
+                        if (attempt >= maxAttempts) {
+                            throw new Error(
+                                "Maximum retry attempts reached. Authentication failed."
+                            );
+                        }
+                    }
+                }
+            }
```

Comment:
In case the error is a different type (not related to bad passphrase), we should throw the original error message to aid in troubleshooting:
```suggestion
            } else {
                throw e;
            }
```